pkg/instrumentation/sdk.go

```diff
@@ -291,12 +294,26 @@ func (i *sdkInjector) addParentResourceLabels(ctx context.Context, uid bool, ns
 			}
 			// parent of ReplicaSet is e.g. Deployment which we are interested to know
 			rs := appsv1.ReplicaSet{}
-			// ignore the error. The object might not exist, the error is not important, getting labels is just the best effort
-			//nolint:errcheck
-			i.client.Get(ctx, types.NamespacedName{
-				Namespace: ns.Name,
-				Name:      owner.Name,
-			}, &rs)
+			nsn := types.NamespacedName{Namespace: ns.Name, Name: owner.Name}
+			backOff := wait.Backoff{Duration: 10 * time.Millisecond, Factor: 1.5, Jitter: 0.1, Steps: 20, Cap: 30 * time.Second} // TODO decide which of these we need and what they should be set to
+
+			checkError := func(err error) bool {
+				// if the error looks like 'ReplicaSet.apps "my-deployment-with-sidecar-f46b479f" not found' ignore it
+				if strings.HasPrefix(err.Error(), "ReplicaSet.apps") && strings.HasSuffix(err.Error(), "not found") {
+					return true
+				}
+				return false
+			}
+
+			getReplicaSet := func() error {
+				return i.client.Get(ctx, nsn, &rs)
+			}
+
+			// use a retry loop to get the Deployment. A single call to client.get fails occasionally
+			err := retry.OnError(backOff, checkError, getReplicaSet)
```

Comment:
Should we use the same retry approach for other objects as well?

Reply:
We could.  Do you have specific places you'd want to do this?  So far this is the only place I've seen failures

Reply:
oh right this is the only place that uses k8s client in this function. I don't know any other place.